### PR TITLE
Try removing attempt to set the C++ standard

### LINKED ISSUE #1460: [BUG] requirement of C++11 is incompatible with Google Test
Google Benchmark insists on being compiled with the standard not exceeding C++11
https://github.com/google/benchmark/blob/974cd5a5c5a78e76ebc50961f4dbf3bf6d4ade4e/CMakeLists.txt#L174

At the same time, Google Benchmark requires Google Test, which is incompatible with C++11 
https://github.com/google/googletest/blob/5b909beeec178f338be997830b6c31a80cda7a93/googletest/include/gtest/internal/gtest-internal.h#L635
```
  typedef ::std::map<std::string, CodeLocation, std::less<>> RegisteredTestsMap;
```
`std::less` without template parameters doesn't compile in C++11.

### PR DESCRIPTION
**Pull request by dmah42**
Fixes #1460 #1462

Let's see what the buildbots make of this.

### CODE REVIEW

**Comment by LebedevRI** [comment]
No, this is wrong.

**Comment by dominichamon** [comment]
> No, this is wrong.

it's just a draft because i want to see what happens... the compiler used should be free to set it's standard. we already had a fallback to c++0x so the only issue i think would be if someone tries to use a compiler that doesn't support c++11.

if we want to require that then setting `CMAKE_CXX_STANDARD` and maybe `REQUIRED` is the way to do that.. but i just want to see how necessary it is.

**Comment by LebedevRI** [comment]
There is absolutely no requirement that a supported compiler version actually happens
to have the right (the one you expect, and everyone expects different things) default standard.

**Comment by dominichamon** [comment]
first test complete: OSX falls back too far beyond C++11.

next up: setting the standard.

**Comment by LebedevRI** [comment]
https://cmake.org/cmake/help/latest/policy/CMP0067.html might be relevant.

**Comment by dominichamon** [comment]
> https://cmake.org/cmake/help/latest/policy/CMP0067.html might be relevant.

nice :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -137,6 +138,9 @@ if (BENCHMARK_BUILD_32_BITS)
   add_required_cxx_compiler_flag(-m32)
 endif()
 
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (LebedevRI) [inline CMakeLists.txt:142]: Probably also want `CMAKE_CXX_EXTENSIONS OFF`

Comment (dominichamon) [reply CMakeLists.txt:142]: yeah.. i'm also eyeing up the check where we:

```
set(EXTRA_CXX_FLAGS "")
if (WIN32 AND "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
  # Clang on Windows fails to compile the regex feature check under C++11
  set(EXTRA_CXX_FLAGS "-DCMAKE_CXX_STANDARD=14")
endif()
```

and wondering how that will end up interacting with this change.

**Comment by dominichamon** [comment]
> > https://cmake.org/cmake/help/latest/policy/CMP0067.html might be relevant.
> 
> nice :)

cmake 3.8+ only .. ubuntu 16.04 is still on 3.5.1: https://launchpad.net/ubuntu/xenial/+source/cmake

**Comment by dominichamon** [comment]
`try_compile` also doesn't support setting the standard in 3.5.1.

so our older platform support requires explicit `--std` flag setting, i think... and for consistency, it suggests we need to do the same in the rest of the project.

until we can drop older platforms, i don't think we can embrace updated cmake norms...

any other ideas, @LebedevRI ?

**Comment by LebedevRI** [comment]
cmake 3.5.1 sounds *really* ancient.
Even debian old-old-stable has 3.7.
I think 3.13 should be an unambiguous dependency.

**Comment by dominichamon** [comment]
ubuntu bionic (the oldest LTS ubuntu we support) is 3.10.2 (https://packages.ubuntu.com/bionic/cmake) so i think we can get past 3.8.

it might mean dropping the 14.04 and 16.04 bots but they are _really_ old at this point and we do have  a policy on dependencies for a reason.

**Comment by LebedevRI** [comment]
Sounds reasonable?

File: CMakeLists.txt [hunk 2]
```diff
@@ -137,6 +139,10 @@ if (BENCHMARK_BUILD_32_BITS)
...
```
Comment (sergiud) [inline CMakeLists.txt:143]: This is not meaningful. MSVC, for instance, does not provide exclusive C++11 support, only C++14 and later: https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170. CMake will probably decay to the newer standard anyway.

**Comment by dominichamon** [comment]
> Sounds reasonable?

image

seems like we might just have had our hand forced...

File: CMakeLists.txt (hunk 1 above)
Comment (powderluv) [reply CMakeLists.txt:143]: yes FWIW we need this check for clang on windows.

**Comment by LebedevRI** [comment]
@dmah42 what's the status here?

**Comment by dmah42** [comment]
> @dmah42 what's the status here?

i don't remember where i got to. let me see if i can resurrect it.

**Comment by dmah42** [comment]
it builds and passes tests but I'm not sure  it's the right thing to do.  I would appreciate an ack.

**Comment by LebedevRI** [comment]
> it builds and passes tests but I'm not sure it's the right thing to do. I would appreciate an ack.

Given that we don't want to just happen to build under whatever the standard is default
for the compiler that is used, but specifically with C++11, this is the right thing to do.

**Review by LebedevRI — APPROVED** [review]
LG

**Comment by dmah42** [comment]
merged.. now let's see if anyone finds an issue on any unusual platforms with it.

**Pull request merged**